pkg/argocd/update.go

```diff
@@ -416,15 +416,17 @@ func marshalParamsOverride(app *v1alpha1.Application, originalData []byte) ([]by
 		}
 
 		if strings.HasPrefix(app.Annotations[common.WriteBackTargetAnnotation], common.HelmPrefix) {
-			images := GetImagesFromApplication(app)
+			images := GetImagesAndAliasesFromApplication(app)
 
 			for _, c := range images {
-				helmAnnotationParamName, helmAnnotationParamVersion := getHelmParamNamesFromAnnotation(app.Annotations, c.ImageName)
```

Comment:
why you need to change this logic? why just not replace it with c.ImageAlias if it is exist ?

Reply:
Thank you for pointing it out :)

There are mainly two reasons for this decision:

1. The code of _getHelmParamNamesFromAnnotation_ it's only used here and anywhere else in the code. I used it at first because it was convenient at the time, but it seems that it may be deprecated if it has no use anymore (I was going to suggest that in another issue later on).
2. Using _GetParameterHelmImageName_ and _GetParameterHelmImageTag_ seems a better option because this two functions uses _normalizedSymbolicName_ function internally, which normalizes the _ImageAlias_ enabling its use in annotations properly because it replaces the "/" character for "_".

If in the end it should be better to leave it as before, just let me know and I'll make revert the changes to use _c.ImageAlias_.

Reply:
I just concerned that you actually changing logic, because inside getHelmParamNamesFromAnnotation here more conditions. I would not change it for now. Also this function is well covered with tests, in case if you are going to use  new logic, you have to cover it with unit tests

Reply:
Yup, you are absolutely right and I should think thrice before changing things.

I'll change it back to the previous statement changing the _c.ImageName_ for _c.normalizedSymbolicName()_ which will work properly with the annotation names because it already replaces the "/" characters for "_". I'll check if the logic works that way and push the changes back.

Thank you for noticing and explaining it :)

Reply:
In order to make the _normalizedSymbolicName_ public to the _update_ package, I'll need to uppercase the first letter and change its references in other parts of the code. Would that be alright?

I should use that function in order to avoid that images names as _"super/app"_ will be used when referencing an annotation name and use "super_app" instead. This will avoid errors referencing annotations with malformed names. I'll give it a try and let me know what do you think.

Reply:
What do you think about idea, pass ContainerImage to getHelmParamNamesFromAnnotation, and just to same refactoring as you did in your PR, but in such case here still will be ability to cover this function with tests

Reply:
I'm not sure if the changes I just pushed are what you meant before.

With this changes, there is no change in logic anywhere and all the current test should cover the same lines. Maybe this solutions works for you?

Reply:
Do you mind to give me write permissions to you fork? want to do small enhancement

Reply:
Sure! I just send the collaboration invitation :)

Reply:
@askhari i have pushed a change

Reply:
@pasha-codefresh , thank you very much for your help. I really appreciate it and with those changes I now understand what you meant by passing the container image. Thank you!

Do you need me to work on some more changes from my side for this PR? I think that with the changes you made, the problem is already solved.

Reply:
@askhari maybe you can help with manual test?

Reply:
Sure! I'll run it on a cluster where we have a test application just for this purpose and will paste here the results to show the validation.

Reply:
Fine for me

Reply:
I see that this is already merged. Thank you for your help @pasha-codefresh .

Still, I attach a test I made to validate that the code is retrieving the information properly when using aliases.
Below there is a screenshot which uses an application definition with a docker image called _labduck_ and using an alias _labbat_ to validate that it is retrieving the values properly from the annotations.

![Captura de pantalla 2024-06-05 a las 17 59 27](https://github.com/argoproj-labs/argocd-image-updater/assets/4053188/01bd8afe-bedf-46d8-8957-892496bf4605)

Again thank you for your help to solve this issue.

Cheers!